src/main/java/org/mitre/inferno/rest/ValidatorEndpoint.java

```diff
@@ -56,115 +49,34 @@ private void createRoutes() {
       res.header("Access-Control-Allow-Origin", "*");
       res.header("Access-Control-Allow-Methods", "GET, POST, PUT, OPTIONS");
       res.header("Access-Control-Allow-Headers", "Access-Control-Allow-Origin, Content-Type");
+      res.type("application/json");
     });
 
     // This responds to OPTIONS requests, used by browsers to "preflight" check CORS requests,
     // with a 200 OK response with no content and the CORS headers above
-    options("*",
-        (req, res) -> {
-          return "";
-        });
+    options("*", (req, res) -> "");
 
     post("/validate",
         (req, res) -> {
           res.type("application/fhir+json");
           return validateResource(req.bodyAsBytes(), req.queryParams("profile"));
         });
 
-    get("/resources",
-        (req, res) -> {
-          res.type("application/json");
-          return resourcesList();
-        });
+    get("/resources", (req, res) -> validator.getResources(), TO_JSON);
 
-    get("/profiles",
-        (req, res) -> {
-          res.type("application/json");
-          return getProfiles();
-        });
+    get("/profiles", (req, res) -> validator.getStructures(), TO_JSON);
 
-    get("/igs",
+    post("/profiles",
```

Comment:
NIT: I think this actually should be singular (e.g. `profile` instead of `profiles`), unless it does accept more than one profile at a time? REST semantics want create operations (which I think this counts as) to be `POST /<singular noun>`, even though it's different from the read operation (which wants `GET /<plural noun>`)

Reply:
Update: I was wrong.